models/snowflake/conversions_by_customer_id.sql

```diff
@@ -0,0 +1,24 @@
+SELECT
+    CASE
+        WHEN events.user_id IS NOT NULL AND events.user_id != '' THEN 'u' || events.user_id -- use event user_id
+        {% if var('use_snowplow_web_user_mapping_table') %}
+            WHEN user_mapping.domain_userid IS NOT NULL THEN 'u' || user_mapping.user_id
+        {% endif %}
+        ELSE 'f' || events.domain_userid
+    END AS customerId,
+    derived_tstamp AS conversionTimestamp,
+    {{ conversion_value() }} AS revenue
+FROM
```

Comment:
Thanks Agnes. I think I understand that you would not recommend querying atomic.events, but I'm a bit confused by the rest. In terms of expense, i'm not sure that it would be overly expensive -as only a few columns are being selected, even if the date range is large. 
Happy to do what you recommend, but just a little confused as to what that is exactly!

Reply:
Hi @agnessnowplow, if you could elaborate on this as discussed on Friday, maybe with some examples or something to help us understand what you are suggesting? That would be very helpful :) Thanks

Reply:
For sure. So I would make it possible for the user to either use the `atomic.events` as a source for conversions as you suggested or they would put their own choice of model as a source within the FROM clause through a variable. The recommended suggestion to use for this would be one of the derived tables of the new ecommerce model that is currently being finalised, most likely this one, which contains the revenue generating transaction events by users: 
<img width="367" alt="image" src="https://user-images.githubusercontent.com/95634439/204624080-f4483c90-6141-4631-801c-33a225221833.png">
If they do that then there is no need to query the event table again just for these transactions, those events will already be available in a much smaller source table. They could also create their own custom incremental model to use as a source, if they wanted to, as I highlighted earlier - as a third option. (Of course they would have to make sure that the data they have is already up-to-date with the range they are querying.) How it could work, for instance, is there would be a new variable called `conversion_source` with atomic and events from the source by default ( `conversion_source: "{{ source('atomic', 'events') }}"`) We would then instruct the user to overwrite this source variable in their own dbt_project.yml if they want to and use their preferred model instead. Let me know if this helps / your thoughts. Thanks!

Reply:
Ahhh I understand now, thanks so much for taking the time to write all this out. That's a good idea, I'll make the necessary changes :)

Reply:
I've made the changes to this and also in the modelling_3.md file in the other repo, hopefully these changes are what you were wanting!